### PR TITLE
🧠 Logic: 📪 `bech32_address/2` predicate

### PR DESCRIPTION
**Pull request by bdeneux**
# Implement #295  : 

## 📝 Purpose

In the `logic` module, purpose an utility function to manipulate bech32 address. 

## 🧪 Expected behavior

`bech32_address/2` is a predicate that convert a bech32 encoded string into base64 bytes and give the address prefix or convert a prefix and base64 encoded bytes to bech32 encoded string. The signature is as follow:

```prolog
bech32_address(-Address, +Bech32)
bech32_address(+Address, -Bech32)
bech32_address(+Address, +Bech32)
```

Where:
  - `Address` is a pair containing HRP (Human-Readable Part) containing the address prefix and the list of integers between 0 and 255 (byte) that represent the base64 encoded bech32 address string. Represented like this : `-(HRP, Address)`
  - `Bech32` is an Atom or string representing the bech32 encoded string address 

## 🎯 Example

```
bech32_address(-(Hrp, Address), 'okp415wn30a9z4uc692s0kkx5fp5d4qfr3ac7sj9dqn').
```

Give : 
```json
{
    "Hrp":     "okp4",
    "Address": "[163,167,23,244,162,175,49,162,170,15,181,141,68,134,141,168,18,56,247,30]",
}
```

### CODE REVIEW

File: x/logic/predicate/address.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import (
+	"context"
+	"fmt"
+
+	bech322 "github.com/cosmos/cosmos-sdk/types/bech32"
+	"github.com/ichiban/prolog/engine"
+	"github.com/okp4/okp4d/x/logic/util"
+)
+
+func Bech32Address(vm *engine.VM, address, bech32 engine.Term, cont engine.Cont, env *engine.Env) *engine.Promise {
```
Comment (ccamel) [inline x/logic/predicate/address.go:12]: Please provide an exhautive description of the function (see: [BankBalances](https://github.com/okp4/okp4d/blob/main/x/logic/predicate/bank.go#L13-L31)).

File: x/logic/predicate/address.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+				return engine.Error(fmt.Errorf("bech32_address/2: %w", err))
+			}
+			return engine.Unify(vm, bech32, util.StringToTerm(bech32Decoded), cont, env)
+		default:
+			return engine.Error(fmt.Errorf("bech32_address/2: you should give at least on instantiated value (Address or Bech32)"))
+		}
+	})
+}
+
+func AddressPairToBech32(addressPair engine.Compound, env *engine.Env) (string, error) {
```
Comment (ccamel) [inline x/logic/predicate/address.go:40]: Should be private I think.

**Review by ccamel — CHANGES_REQUESTED** [review]
The implementation is quite good! Thanks for the job. ☺️

However, I have a few concerns about the error messages, their consistency (especially in relation to other predicates), and their formalization. But I think we can address this in a second step.

It would also be helpful to provide a thorough description of this predicate, as the generated documentation will be very useful for understanding the predicates supported by the module.

File: x/logic/predicate/address.go (hunk 1 above)
Comment (bdeneux) [reply x/logic/predicate/address.go:12]: Oups, I forgot write it 🫢.

**Review by ccamel — APPROVED** [review]
Great job! Thanks :sunglasses:

**Review by amimart — APPROVED** [review]
Great! 💪

**Pull request merged**